### ISSUE TITLE
Allow changing vsync on a Window

### ISSUE THREAD

**Issue by bitshifter** [rust-windowing/glutin#693]
Perhaps there is a good reason for not supporting this, but many APIs do allow changing the present interval on the fly and it's useful to be able to toggle vsync without recreating the Window, device context and all resources. Perhaps a return value if the operation was supported and successful would suffice.

Something like:

``` rust
impl Window {
  fn set_vsync(enabled: bool) -> Result<(), SomeError> {}
  // or even 
  fn set_present_interval(interval: PresentInterval) -> Result<(), SomeError> {}
```

Does this make sense for glutin?

**Comment by Aceeri** [comment]
While we are talking about vsync, what about changing vsync from a bool to an i8? That way you can pass in a -1 for adaptive vsync, 0 for no vsync, and 1 for vsync.

As for allowing changes in runtime, it looks like that the implementations for vsync are all setting SwapInterval at initialization right now.

**Comment by adambadawy** [comment]
> That way you can pass in a -1 for adaptive vsync, 0 for no vsync, and 1 for vsync.

An `enum` seems more appropriate for a setting like that.

**Comment by fkaa** [comment]
VSync doesn't sound fun at all. From GLFW's [glfwSwapInterval(int interval)](http://www.glfw.org/docs/latest/group__context.html#ga6d4e0cdf151b5e579bd67f13202994ed):

> Contexts that support either of theWGL_EXT_swap_control_tear andGLX_EXT_swap_control_tear extensions also accept negative swap intervals, which allow the driver to swap even if a frame arrives a little bit late.
> [...]
> This function is not called during context creation, leaving the swap interval set to whatever is the default on that platform. This is done because some swap interval extensions used by GLFW *do not allow the swap interval to be reset to zero once it has been set to a non-zero value*.

**Comment by kchibisov** [comment]
Fixed in #1439.
